### PR TITLE
File action to open PDFs with richdocuments when files_pdfviewer is enabled

### PR DESCRIPTION
**Pull request by elzody**
* Resolves: #3621 
* Target version: main

### Summary
Since 28, when the files_pdfviewer app is enabled, there is no open to open a PDF using richdocuments. This PR fixes that, using the files API to register a file action. By default, when you click on the file, it will still use the files_pdfviewer when enabled.

### Checklist

- [x] Code is properly formatted
- [x] Sign-off message is added to all commits
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: img/app.svg
Comment (juliushaertl) [inline img/app.svg:1]: We should rather add a separate app-dark.svg file. Unfortunately the apps management UI currently requires the main app icon to be white.

**Review by juliushaertl — COMMENTED** [review]
Small comment on the app icon, otherwise very nice and good to get in 👍

File: img/app.svg
Comment (elzody) [reply img/app.svg:1]: Ah, that explains why there are already two of them then! I will revert it :)

**Review by juliushaertl — APPROVED** [review]

**Comment by juliushaertl** [comment]
@elzody Would be quite nice if we could follow up with a small cypress test for this to avoid further regressions

**Pull request merged**